src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int NUMBER_OF_FIGURE = 6;
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[NUMBER_OF_FIGURE];
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        for (int i = 0; i < NUMBER_OF_FIGURE / 2; i++) {
```

Comment:
There is no need to divide the 'NUMBER_OF_FIGURE' by 2 during the iteration. Instead, you can generate all figures in a single loop and decide within the loop whether to call 'getRandomFigure()' or 'getDefaultFigure()' based on the current index.